### PR TITLE
ci: reenable mac and enable aarch64 builds

### PR DESCRIPTION
**Pull request by platten**
Signed-off-by: Paul Pietkiewicz <paul@profian.com>

### CODE REVIEW

**Review by bstrie — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -57,11 +51,45 @@ jobs:
...
+        run: echo "::set-output name=toolchain_version::$(grep channel  rust-toolchain.toml | cut -d\" -f 2)"
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ steps.toolchain_version.outputs.toolchain_version }}
+          target: ${{ matrix.platform.target }}
+          profile: minimal
+      - name: Setup Rust toolchain
+        run: >
+            rustup override unset || :
+            && rustup show
```
Comment (rvolosatovs) [inline .github/workflows/release.yml:76]: What does this do?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -57,11 +51,45 @@ jobs:
         name: enarx-${{ matrix.platform.target }}-oci
         path: enarx-${{ matrix.platform.target }}-oci
 
+  aarch64-build:
```
Comment (rvolosatovs) [inline .github/workflows/release.yml:54]:
```suggestion
  build-aarch64:
```

And original `build` should be renamed to `build-x86_64`

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -57,11 +51,45 @@ jobs:
...
+          command: build
+          args: --release --target ${{ matrix.platform.target }}
+      - name: Rename artifact
+        run: mv target/${{ matrix.platform.target }}/release/enarx enarx-${{ matrix.platform.target }}
+      - uses: actions/upload-artifact@v3
+        with:
+          name: enarx-${{ matrix.platform.target }}
+          path: enarx-${{ matrix.platform.target }}
+
   windows-build:
```
Comment (rvolosatovs) [inline .github/workflows/release.yml:88]:
```suggestion
  build-windows:
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -84,49 +112,47 @@ jobs:
...
+    - run: lipo -create ./enarx-aarch64-apple-darwin ./enarx-x86_64-apple-darwin -output ./enarx-universal-darwin
+    - uses: actions/upload-artifact@v3
+      with:
+        name: enarx-universal-darwin
+        path: enarx-universal-darwin
 
   test-bin:
-    needs: build
-    # needs: universal-binary
+    needs: universal-binary
```
Comment (rvolosatovs) [inline .github/workflows/release.yml:144]: Add explicit dependencies on `build-x86_64` and `build-aarch64`

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -162,12 +188,11 @@ jobs:
     - run: docker load < enarx-${{ matrix.platform.target }}-oci
     # TODO: Attempt to run `enarx info` within the loaded container
 
-  Regenerate-BOM:
-    uses: ./.github/workflows/sbom_manifest_action.yml
+  # TODO: redevelop SBOM generation
```
Comment (rvolosatovs) [inline .github/workflows/release.yml:191]: Is there an issue about this? If so, please add a reference. Otherwise, file one and add a reference

**Review by rvolosatovs — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (platten) [reply .github/workflows/release.yml:76]: Unsets the override in cargo/config and provides a workaround in a component install bug in the github action. Toolchain and target wanted to default to another target without this defined.

File: .github/workflows/release.yml (hunk 5 above)
Comment (platten) [reply .github/workflows/release.yml]: Referenced in issue #1954

**Review by rvolosatovs — APPROVED** [review]

**Review by bstrie — APPROVED** [review]

**Pull request merged**